app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt

```diff
@@ -42,11 +41,23 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
         ActivityResultContracts.RequestMultiplePermissions()
     ) { isGranted: Map<String, Boolean> ->
         if (isGranted.all { it.value }) {
-            binding.swiperefresh.isRefreshing = true
-            setDefaultSubscriptionId()
-            onRefresh()
+            // phone permissions granted
+            if(isGranted.containsKey(CALL_PHONE) || isGranted.containsKey(READ_PHONE_STATE)){
+                binding.swiperefresh.isRefreshing = true
+                setDefaultSubscriptionId()
+                onRefresh()
+            }
+
         } else {
-            showSnackbar(R.string.permissions_required)
+            // notification permission denied
+            if(isGranted.containsKey(POST_NOTIFICATIONS)){
+                showSnackbar(R.string.notification_permission_denied)
+            }
+
+            // phone permissions denied
+            if(isGranted.containsKey(CALL_PHONE) || isGranted.containsKey(READ_PHONE_STATE)){
+                showSnackbar(R.string.permissions_required)
+            }
```

Comment:
I'd prefer something like this:

```suggestion
           val message = when {
             isGranted.containsKey(POST_NOTIFICATIONS) -> R.string.notification_permission_denied
             ....
             else -> // throw AssertionError
             }
          showSnackbar(message )   
```

Reply:
resolved by latest commit